sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -925,8 +925,22 @@ object IntervalUtils {
    * @throws ArithmeticException If numeric overflow occurs
    */
   def periodToMonths(period: Period): Int = {
+    periodToMonths(period, YearMonthIntervalType.YEAR, YearMonthIntervalType.MONTH)
+  }
+
+  def periodToMonths(period: Period, startField: Byte, endField: Byte): Int = {
     val monthsInYears = Math.multiplyExact(period.getYears, MONTHS_PER_YEAR)
-    Math.addExact(monthsInYears, period.getMonths)
+    val months = Math.addExact(monthsInYears, period.getMonths)
+    (startField, endField) match {
```

Comment:
the general rule is: we should "round" to the rightmost unit, but not the leftmost unit, which is too risky. I think we can simplify the code here
```
if (endField == YEAR) {
  code to truncate to year
} else {
  months
}
```

Reply:
Done